vatvalidate/validate.py

```diff
@@ -0,0 +1,50 @@
+def _modulus_9755(vat_digits: list[int], use_9755: bool = False) -> bool:
+    """
+    Applies the modulus 97 algorithm or the modulus 9755 algorithm to the provided
+    vat digits.
+    """
+    # Multiply first seven digits by weights from 8 to 2 and sum them
+    check_digits = 0
+    for i in range(7):
+        check_digits += vat_digits[i] * (8 - i)
+
+    # Optionally add 55 (modulus 9755 algorithm)
+    if use_9755:
+        check_digits += 55
+
+    # Subtract 97 until we get a negative number, then take the absolute value
+    while check_digits > 0:
+        check_digits -= 97
+    check_digits = abs(check_digits)
```

Comment:
similarly here - it doesn't appear quicker, but I feel like it makes it less easy to read

Reply:
It looks 6x faster to me

```python
     ...: def get_check_digits(check_digits: int) -> int:
     ...:     while check_digits > 0:
     ...:         check_digits -= 97
     ...:     return abs(check_digits)
     ...:
     ...:
     ...: def get_check_digits_2(check_digits: int) -> int:
     ...:     return abs(check_digits - 97 * ceil(check_digits / 97))
     ...:

In [249]: %%timeit
     ...: get_check_digits(randint(0, 20000))
     ...:
     ...:
2.24 µs ± 273 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [250]: %%timeit
     ...: get_check_digits_2(randint(0, 20000))
     ...:
     ...:
365 ns ± 2.63 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
ah good - I think I was comparing the previous timings rather than the ones i generated for this code :facepalm: cos looking again I get similar results to you

Reply:
https://github.com/Beauhurst/vatvalidate/pull/1/commits/c83ba36312fa25b25e1d1ba85e278872061c08e1